src/Donations/Micros/Authentication.tsx

```diff
@@ -43,16 +44,33 @@ function Authentication({}: Props): ReactElement {
   const [openVerifyEmailModal, setopenVerifyEmailModal] = React.useState(false);
 
   const loadUserProfile = async () => {
-    if ((user && user.email_verified) || queryToken) {
+    let token;
+    if (queryToken) {
+      const jwtToken = parseJwt(queryToken);
+      const isTokenValid =
+        new Date(jwtToken.exp * 1000).getTime() > new Date().getTime();
```

Comment:
I wonder if we cannot use the same solution with `jwt-decode` as in planet-webapp by @sarveshpro with his commit https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1381/commits/257057b0f446e4cd23c7caff6e4763405542bd76 for https://github.com/Plant-for-the-Planet-org/planet-webapp/pull/1381?

Reply:
Besides this self made parsing crashes e.g. with `token=test`:
```
Uncaught (in promise) TypeError: Cannot read properties of undefined (reading 'replace')
    at parseJwt (getFormattedNumber.ts?1d7b:97:26)
    at _callee$ (Authentication.tsx?8848:50:32)
```